File: src/com/kedzie/vbox/api/IHostNetworkInterface.java
package com.kedzie.vbox.api;

import java.util.HashMap;
import java.util.Map;

import android.os.Parcel;
import android.os.Parcelable;

import com.kedzie.vbox.api.jaxb.HostNetworkInterfaceMediumType;
import com.kedzie.vbox.api.jaxb.HostNetworkInterfaceStatus;
import com.kedzie.vbox.api.jaxb.HostNetworkInterfaceType;
import com.kedzie.vbox.soap.KSOAP;
import com.kedzie.vbox.soap.VBoxSvc;

public interface IHostNetworkInterface extends IManagedObjectRef, Parcelable {
	public static final String BUNDLE = "hostNetwork";
	static final ClassLoader LOADER = IHostNetworkInterface.class.getClassLoader();
	
	public static Parcelable.Creator<IHostNetworkInterface> CREATOR = new Parcelable.Creator<IHostNetworkInterface>() {
		@Override
		public IHostNetworkInterface createFromParcel(Parcel in) {
			VBoxSvc vmgr =  in.readParcelable(LOADER);
			String id = in.readString();
			Map<String, Object> cache = new HashMap<String, Object>();
			in.readMap(cache, LOADER);
			return (IHostNetworkInterface) vmgr.getProxy(IHostNetworkInterface.class, id, cache); 
		}
		
		@Override
		public IHostNetworkInterface[] newArray(int size) {
			return new IHostNetworkInterface[size];
		}
	};

	@KSOAP(cacheable=true) public String getName();
	
	@KSOAP(cacheable=true) public String getId();
	
	@KSOAP(cacheable=true) public String getNetworkName();
	
	@KSOAP(cacheable=true) public boolean getDHCPEnabled();

	@KSOAP(cacheable=true) public String getIPAddress();
	
	@KSOAP(cacheable=true) public String getNetworkMask();

	@KSOAP(cacheable=true) public boolean getIPV6Supported();
	
	@KSOAP(cacheable=true) public String getIPV6Address();
	
	@KSOAP(cacheable=true) public int getIPV6NetworkMaskPrefixLength();
	
	@KSOAP(cacheable=true) public String getHardwareAddress();
	
	@KSOAP(cacheable=true) public HostNetworkInterfaceMediumType getMediumType();
	
	@KSOAP(cacheable=true) public HostNetworkInterfaceStatus getStatus();
	
	@KSOAP(cacheable=true) public HostNetworkInterfaceType getInterfaceType();
	
	public void enableStaticIPConfig(@KSOAP("IPAddress") String ipAddress, @KSOAP("networkMask") String networkMask);
	
	public void enableStaticIPConfigV6(@KSOAP("IPV6Address") String ipv6Address, @KSOAP(type="unsignedInt", value="IPV6NetworkMaskPrefixLength") int ipv6NetworkMaskPrefixLength);
	
	public void enableDynamicIPConfig();
	
    @KSOAP(cacheable=true) public boolean DHCPRediscover();
}

File: vbox-app/src/main/java/com/kedzie/vbox/api/IHostNetworkInterface.kt
package com.kedzie.vbox.api

import android.os.Parcelable
import com.kedzie.vbox.api.jaxb.HostNetworkInterfaceMediumType
import com.kedzie.vbox.api.jaxb.HostNetworkInterfaceStatus
import com.kedzie.vbox.api.jaxb.HostNetworkInterfaceType
import com.kedzie.vbox.soap.Cacheable
import com.kedzie.vbox.soap.Ksoap
import com.kedzie.vbox.soap.KsoapProxy

@KsoapProxy
@Ksoap
interface IHostNetworkInterface : IManagedObjectRef, Parcelable {

    companion object {
        const val BUNDLE = "hostNetwork"
    }

    @Cacheable( "Name")
	suspend fun getName(): String

    @Cacheable("Id")
	suspend fun getId(): String

    @Cacheable("NetworkName")
	suspend fun getNetworkName(): String

    @Cacheable("DHCPEnabled")
	suspend fun getDHCPEnabled(): Boolean

    @Cacheable("IPAddress")
	suspend fun getIPAddress(): String

    @Cacheable("NetworkMask")
	suspend fun getNetworkMask(): String

    @Cacheable("IPV6Supported")
    suspend fun getIPV6Supported(): Boolean

    @Cacheable("IPV6Address")
    suspend fun getIPV6Address() :String

    @Cacheable("IPV6NetworkMaskPrefixLength")
    suspend fun getIPV6NetworkMaskPrefixLength(): Int

    @Cacheable("HardwareAddress")
	suspend fun getHardwareAddress(): String

    @Cacheable("MediumType")
	suspend fun getMediumType(): HostNetworkInterfaceMediumType

    @Cacheable("Status")
	suspend fun getStatus(): HostNetworkInterfaceStatus

    @Cacheable("InterfaceType")
	suspend fun getInterfaceType(): HostNetworkInterfaceType

    @Cacheable("DHCPRediscover")
    suspend fun DHCPRediscover(): Boolean
}